app/uk/gov/hmrc/hecstubs/util/ValidationUtils.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hecstubs.util
+
+import cats.data.Validated.Valid
+import cats.data.{Validated, ValidatedNel}
+import uk.gov.hmrc.hecstubs.models.accountOverviewDetails.ErrorResult
+import uk.gov.hmrc.hecstubs.models.accountOverviewDetails.InvalidCode.InvalidCorrelationId
+
+import java.util.UUID
+import scala.util.Try
+
+object ValidationUtils {
+  def correlationIdValidation(correlationId: String): ValidatedNel[ErrorResult, String] = Try(
+    UUID.fromString(correlationId)
+  ).toOption match {
+    case Some(id) => Valid(id.toString)
+    case None     =>
+      Validated.invalidNel(
+        ErrorResult(InvalidCorrelationId, "Submission has not passed validation. Invalid header CorrelationId.")
+      )
+  }
```

Comment:
maybe we can directly match on the `Try` result instead of converting to `Option`? Something like -
```
def correlationIdValidation(correlationId: String): ValidatedNel[ErrorResult, String] = Try(
    UUID.fromString(correlationId)
  ) match {
    case Failure(_) => Validated.invalidNel(
      ErrorResult(InvalidCorrelationId, "Submission has not passed validation. Invalid header CorrelationId.")
    )
    case Success(id) =>Valid(id.toString)
  }
```

Reply:
It's same thing :) anyhow we have to do pattern matching

Reply:
I usually prefer Success and failure when i have to use the exception otherwise i opt for optional value. No worries, i can change that